File: wordsmanship.rb
# http://programmingpraxis.com/2014/11/25/thou-impertinent-urchin-faced-miscreant/

def wordsmanship(code)
  column_1 = ["integrated","total","systematized","parallel","functional",
              "responsive","optional","synchronized","compatible","balanced"]

  column_2 = ["management","organizational","monitored","reciprocal","digital",
              "logistical","transitional","incremental","third-generation",
              "policy"]

  column_3 = ["options","flexibility","capability","mobility","programming",
              "concept","time-phase","projection","hardware","contingency"]
  case 
  when !(code == code.to_i)
    "#{column_1.sample} #{column_2.sample} #{column_3.sample}"
  when code > 999
    "#{column_1.sample} #{column_2.sample} #{column_3.sample}"
  when code < 0
    "#{column_1.sample} #{column_2.sample} #{column_3.sample}"
  else
    val = code.to_s.split(//).map { |digit| digit.to_i }
    "#{column_1[val[0]]} #{column_2[val[1]]} #{column_3[val[2]]}"
  end
end    

p wordsmanship(123)
p wordsmanship("horse")
p wordsmanship(1021)

File: wordsmanship.swift
// http://programmingpraxis.com/2014/11/25/thou-impertinent-urchin-faced-miscreant/

func wordsmanship(code:Int) -> String {
  let column1 = ["integrated","total","systematized","parallel","functional",
                  "responsive","optional","synchronized","compatible","balanced"]
  let column2 = ["management","organizational","monitored","reciprocal","digital",
                  "logistical","transitional","incremental","third-generation", 
                  "policy"]
  let column3 = ["options","flexibility","capability","mobility","programming",
                  "concept","time-phase","projection","hardware","contingency"]
  if code > 999 || code < 0 {
    return "(Enter a three digit code between 000 and 999!)"
  } else {
    var value = code
    var digits:[Int] = []
    while value > 0 {
      digits.append(value % 10)
      value /= 10
    }
    let colIndices = reverse(digits)
    let firstWord = column1[colIndices[0]]
    let secondWord = column2[colIndices[1]]
    let thirdWord = column3[colIndices[2]]
    return "\(firstWord) \(secondWord) \(thirdWord)"
  }
}

println(wordsmanship(123))
